plugins/providers/virtualbox/cap/configure_disks.rb

```diff
@@ -167,20 +280,47 @@ def self.create_disk(machine, disk_config)
         #  device = disk_info[3]
         #
         # @param [Vagrant::Machine] machine
+        # @param [VagrantPlugins::ProviderVirtualBox::Model::StorageController] controller -
+        # the storage controller to use
         # @return [Hash] dsk_info - The next available port and device on a given controller
-        def self.get_next_port(machine)
-          vm_info = machine.provider.driver.show_vm_info
-          dsk_info = {device: "0", port: "0"}
-
-          disk_images = vm_info.select { |v| v.include?("ImageUUID") && v.include?("SATA Controller") }
-          used_ports = disk_images.keys.map { |k| k.split('-') }.map {|v| v[2].to_i}
-          next_available_port = ((0..(MAX_DISK_NUMBER-1)).to_a - used_ports).first
+        def self.get_next_port(machine, controller)
+          dsk_info = {}
+
+          if controller.storage_bus == "SATA"
+            used_ports = controller.attachments.map { |a| a[:port].to_i }
+            next_available_port = ((0..(controller.maxportcount - 1)).to_a - used_ports).first
+
+            dsk_info[:port] = next_available_port.to_s
+            dsk_info[:device] = "0"
+          elsif controller.storage_bus == "IDE"
+            # IDE Controllers have primary/secondary devices, so find the first port
+            # with an empty device
+            (0..(controller.maxportcount - 1)).each do |port|
+              # Skip this port if it's full
+              port_attachments = controller.attachments.select { |a| a[:port] == port.to_s }
+              next if port_attachments.count == 2
```

Comment:
The `2` should be assigned to a descriptive constant, and the constant used here instead.

Reply:
I moved this into an reader in the StorageController class, which gets set when the storage controller is initialized.